src/psyclone/profiler.py

```diff
@@ -169,12 +175,24 @@ def __init__(self, children=None, parent=None):
         # change every time gen() is called).
         self._var_name = NameSpaceFactory().create().create_name("profile")
 
-        # Name of the region. In general at constructor time we might not
-        # have a parent subroutine or a child for the kernel, so we leave
-        # the name empty for now. The region and module names are set the
-        # first time gen() is called (and then remain unchanged).
-        self._region_name = None
+        # Name of the region. In general at constructor time we might
+        # not have a parent subroutine or a child for the kernel, so
+        # the name is left empty, unless explicitly provided by the
+        # user. If not names are provided here then the region and
+        # module names are set the first time gen() is called (and
+        # then remain unchanged).
         self._module_name = None
+        self._region_name = None
+        if name:
+            if not isinstance(name, tuple) or not len(name) == 2 or \
```

Comment:
pylint complains about this if statement (too many booleans).

Reply:
I've suppressed the warnings as I don't see how the code would be better if it were split.